src/audio-view.js

```diff
@@ -0,0 +1,193 @@
+    Promise.all(this._promises)
+      // Don't listen to events until we've loaded all sounds
+      .then(() => this.store.on(SculptureStore.EVENT_CHANGE, this._handleChanges.bind(this)))
+      .then(() => callback(null))
+      .catch(callback.bind(null));
+  }
+
+  // Traverses sound config objects and replaces nodes with valid, loaded, sounds
+  _traverse(node) {
+    for (let key in node) {
+      const value = node[key];
+      let sound;
+      if (typeof value === 'string') sound = node[key] = new Sound({url: value});
+      else if (value instanceof Sound) sound = value;
+      if (sound) this._loadSound(sound);
+      else this._traverse(value);
+    }
+  }
+
+  // Returns a promise to provide a comple, usable sound.
+  _loadSound(sound) {
+    const promise = sound.load();
+    this._promises.push(promise);
+    return promise;
+  }
+
+  _handleChanges(changes) {
+    if (this._animating) {
+      return;
+    }
+
+    if (changes.currentGame === GAMES.HANDSHAKE) this.sounds.alone.ambient.play();
+
+    this._handleHandshakeChanges(changes);
+    this._handleStatusChanges(changes);
+    this._handleLightChanges(changes);
+  }
+
+  _handleHandshakeChanges(changes) {
+    if (changes.handshakes) {
+      // Did someone shake my hand?
+      if (changes.handshakes[this.config.username]) {
+        this.sounds.alone.ambient.stop();
+        this.sounds.alone.handshake.play();
+      }
+    }
+  }
+
+  _handleStatusChanges(changes) {
+    if (changes.status) {
+      let statusSounds;
+
+      if (this.store.isPlayingMoleGame) {
+        statusSounds = {
+          [SculptureStore.STATUS_SUCCESS]: this.sounds.mole.success,
+          [SculptureStore.STATUS_FAILURE]: this.sounds.mole.failure
+        };
+      }
+      if (this.store.isPlayingSimonGame) {
+        statusSounds = {
+          [SculptureStore.STATUS_SUCCESS]: this.sounds.simon.success,
+          [SculptureStore.STATUS_FAILURE]: this.sounds.simon.failure
+        };
+      }
+
+      const statusSound = statusSounds[changes.status];
+      if (statusSound) statusSound.play();
+    }
+  }
+
+  _handleLightChanges(changes) {
+    const lightChanges = changes.lights;
+    if (!lightChanges || !this.store.isReady) {
+      return;
+    }
+    
+    if (this.store.isPlayingMoleGame) {
+      const lightArray = this.lightArray;
+      for (let stripId of Object.keys(lightChanges)) {
+        const panels = lightChanges[stripId].panels;
+        for (let panelId of Object.keys(panels)) {
+          const panelChanges = panels[panelId];
+          if (panelChanges.intensity > 90) {
+            this.sounds.mole.panels[stripId][panelId].play();
+          }
+          if (panelChanges.hasOwnProperty("active")) {
+            if (panelChanges.active) {
+              const molegame = this.store.currentGameLogic;
+              const moledata = this.store.data.get('mole');
+              const currentTarget = molegame.getTargetPanels(moledata.get('targetIndex'));
```

Comment:
Rather than doing this and adding a dependency on the structure of the data within the mole game logic, could you create a public property in mole game logic to get the current set of target panels based on the target index?